feature-libs/user/account/components/login/dom-change.directive.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Output,
+} from '@angular/core';
+
+@Directive({
+  selector: '[cxButtonAdded]',
+})
+export class DomChangeDirective implements OnDestroy {
+  private changes: MutationObserver;
+
+  @Output()
+  public cxButtonAdded = new EventEmitter();
+
+  constructor(private elementRef: ElementRef) {
+    this.changes = new MutationObserver((mutations: MutationRecord[]) => {
+      mutations
+        .filter((mutation) => mutation.target?.nodeName === 'BUTTON')
```

Comment:
Maybe we can pass in the filter for mutations as a directive parameter. That way we can reuse this directive for whatever DOM changes we need to check for.

Reply:
Better yet, can we pass in the target to get the changed mutation directly? Something like this:
```
[cxOnDomChange]="
  'cx-navigation-ui nav ul li:first-child button', // target
  callbackFn(target) // returns target element so we can modify it without using querySelector
"
```